src/gl/canvas2dRe.re

```diff
@@ -64,7 +66,9 @@ module LineJoin: LineJoinType = {
   let miter : t = "miter";
 };
 
-
+type style =
+  | String string
+  | Gradient gradientT;
```

Comment:
Could you try converting this into a GADT and using it as a phantom argument instead? See https://bloomberg.github.io/bucklescript/Manual.html#_phantom_arguments_and_ad_hoc_polymorphism. I think this should result in cleaner codegen. Another alternative is to use casting functions , e.g. `external styleFromString : string => style = "%identity"`